guide/blueprints/config-files.md

```diff
@@ -0,0 +1,78 @@
+This looks for the given (fully qualified) resource on the Brooklyn classpath.
+
+The destination for the file upload is specified in the entity's configuration. Note that the backslash character 
+(\\) must be escaped when using Windows. For example: `C:\\install7zip.ps1`.
+
+Files can be uploaded at several stages of an entity's lifecycle:
+
+| Config key             | Copied before lifecycle phase | Templated |
+|------------------------|-------------------------------|-----------|
+| `files.preinstall`     | Pre-install                   | ✕         |
+| `files.install`        | Install                       | ✕         |
+| `files.customize`      | Customize                     | ✕         |
+| `files.runtime`        | Launch                        | ✕         |
+| `templates.preinstall` | Pre-install                   | ✓         |
+| `templates.install`    | Install                       | ✓         |
+| `templates.customize`  | Customize                     | ✓         |
+| `templates.runtime`    | Launch                        | ✓         |
+
+
+### Template substitutions
+
+Templated files (be they configuration files or scripts) gives a powerful way to inject dependent 
+configuration when installing an entity (e.g. for customising the install, or for referencing the
+connection details of another entity). Available substitutions are:
+
+| Substitution              | Effect                                                             |
+|---------------------------|--------------------------------------------------------------------|
+| `${config['key']}`        | Equivalent to `entity.config().get(key)`                           |
+| `${attribute['key']}`     | Equivalent to `entity.sensors().get(key)`                          |
+| `${mgmt['key']}`          | Loads the value for `key` from the management context's properties |
+| `${entity.foo}`           | FreeMarker calls `getFoo` on the entity                            |
+| `${driver.foo}`           | FreeMarker calls `getFoo` on the entity's [driver](http://brooklyn.apache.org/v/latest/java/entity.html#things-to-know) |
+| `${location.foo}`         | FreeMarker calls `getFoo` on the entity's location                 |
+| `${javaSysProps.foo.bar}` | Loads the system property named `foo.bar`                          |
+
+Additional substitutions can be given per-entity by setting the `template.substitutions` key. For example,
+to include the address of an entity called db:
+
+    brooklyn.config
+      template.substitutions:
+        databaseAddress: $brooklyn:entity("db").attributeWhenReady("host.address")
+
+The value can be referenced in a template with `${databaseAddress}`.
+
+
+### Example
+
+    files.preinstall:
+      # Reference a fixed resource
+      classpath://com/acme/installAcme.ps1: C:\\acme\installAcme.ps1
+      # Inject the source from a config key
+      $brooklyn:config("acme.conf"): C:\\acme\acme.conf
```

Comment:
I'd also like to see described how to install files to INSTALL_DIR and/or RUN_DIR - do I just omit the path component?